### PR TITLE
Upload file

### PR DESCRIPTION
**Pull request by rowlando**
I've put together the minimal amount of code to upload a file to /api/v1/upload end-point with a basic spec.

Lots more work to do but I wanted to get something in the main trunk for feedback before making more increments.

You'll need to run migrations as there's a new DB table that will ultimately record each upload.

### CODE REVIEW

**Comment by doctorpod** [comment]
So this takes in a file and stores it on the file system. Weren't we going to have a separate app which does the file uploading, processing and then posting to this app via some create-type endpoint? Apologies if I got this wrong - we've had quite a few discussion and I have the memory of a common house fly ;)

**Comment by rowlando** [comment]
I agree that the following concerns should be in a separate app:

- uploading an excel file
- temp storage of excel file
- summary of private fields / public fields and any other helpful info
- processing data and reporting errors
- POSTing to endpoint in API::App that updates database

## What is a separate app?
To get to a shared understanding about what we mean by a separate app, I think the options are:

**multiple apps in the same codebase**
```
/lib/api/app
/lib/admin/app
```

with a config.ru running these apps, e.g.

```ruby
run API::App
run Admin:App
```

**separate codebases per app**
This will lead to cleaner code, fasting running tests and all the other good stuff. But it maybe premature in the early days whilst we're not sure what the common code is.

_Sorry for the ramble_

What do you think?

**Comment by doctorpod** [comment]
I have a dream! Sorry I briefly became a well know human rights activist. I see a nice lean Sinatra app with a handful of pages and some nicely modularise bit of functionality for the file processing app --- but of course it could equally be Django ;)

**Comment by rowlando** [comment]
[Review pull request 8](https://trello.com/c/53SdYULO/39-review-pull-request-8)

File: config.ru [hunk 1]
```diff
@@ -4,4 +4,10 @@ $LOAD_PATH.unshift("#{File.dirname(__FILE__)}/lib")
 require 'lib/env'
 require 'lib/mediators'
 
-run API::App
+map '/' do
```
Comment (doctorpod) [inline config.ru]: For consistency, maybe map this to `/api` and remove the `prefix :api` from `lib/api/app.rb`

**Pull request merged**